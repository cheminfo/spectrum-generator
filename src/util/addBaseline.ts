import type { Data1D } from '../../types/Data1D';

type baseLineFn = (x: number) => number;

export default function addBaseline(data: Data1D, baselineFct: baseLineFn) {
  if (!baselineFct) return data;
  let xs = data.x;
  let ys = data.y;
  for (let i = 0; i < xs.length; i++) {
    ys[i] += baselineFct(xs[i]);
  }
  return data;
}
