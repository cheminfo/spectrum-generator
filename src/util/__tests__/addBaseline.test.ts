import addBaseline from '../addBaseline';

test('addBaseline', () => {
  let corrected = addBaseline({ x: [1, 2, 3], y: [2, 3, 4] }, (x) => 2 * x);
  expect(corrected).toMatchSnapshot();
});
