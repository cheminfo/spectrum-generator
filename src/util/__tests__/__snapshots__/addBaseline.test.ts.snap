// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addBaseline 1`] = `
Object {
  "x": Array [
    1,
    2,
    3,
  ],
  "y": Array [
    4,
    7,
    10,
  ],
}
`;
